prov/efa/src/rdm/rxr_env.c

```diff
@@ -221,26 +207,78 @@ void rxr_env_define()
 			"The maximum number of bytes that will be eagerly sent by inflight messages uses runting read message protocol (Default 307200).");
 }
 
-/**
- * @brief Initialize the variables in rxr_env.
- * This function uses the g_device_list initialized in
- * efa_device_list_initialize(), thus it must be called
- * after efa_device_list_initialize().
- */
-void rxr_env_initialize()
+
+/* This function specifies the default value for using the
+ * device's RDMA capability. It also gives the help message
+ * about whether the environment variable FI_EFA_USE_DEVICE_RDMA
+ * can be set or not. */
+void rxr_env_define_use_device_rdma()
 {
+	char *str = "";
 	uint32_t vendor_part_id = g_device_list[0].ibv_attr.vendor_part_id;
 
+	/* This is the default scenario. This could be overwritten
+	 * later by the user when fi_param_get_int() is called.
+	 * By default we would not want to use device RDMA in EFA
+	 * gen-0 and gen-1 because it wasn't shipped with RDMA
+	 * capability. Starting from EFA gen-2, if the device
+	 * supports RDMA, use it. */
 	if (vendor_part_id == 0xefa0 || vendor_part_id == 0xefa1) {
-		rxr_env.use_device_rdma = 0;
+		use_device_rdma = 0;
 	} else {
-		rxr_env.use_device_rdma = 1;
+		use_device_rdma = 1;
 		if (!(g_device_list[0].efa_attr.device_caps & EFADV_DEVICE_ATTR_CAPS_RDMA_READ)) {
 			EFA_WARN(FI_LOG_FABRIC,
 				"EFA device with vendor_part_id 0x%x has no rdma-read capability.\n", vendor_part_id);
-			rxr_env.use_device_rdma = 0;
+			use_device_rdma = 0;
```

Comment:
Do not use global variable. Get the default value from `g_device_list[0].ibv_attr.vendor_part_id` in `rxr_env_get_use_device_rdma`.